server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -672,14 +677,15 @@ void startWrite() {
             indexWriter.getConfig().setMergePolicy(NO_MERGE_POLICY);
         }
 
-        void prepareCommit(String nodeId, long currentTerm, long lastAcceptedVersion) throws IOException {
+        void prepareCommit(String nodeId, long currentTerm, long lastAcceptedVersion, int oldestIndexVersion) throws IOException {
```

Comment:
Nit: could we keep this as a `Version` as long as possible?

```suggestion
        void prepareCommit(String nodeId, long currentTerm, long lastAcceptedVersion, Version oldestIndexVersion) throws IOException {
```